agent/src/main.rs

```diff
@@ -57,6 +55,22 @@ async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>
         run_metrics_server().await.unwrap();
     }));
 
+    let discovery_handler_map = Arc::new(Mutex::new(HashMap::new()));
+    let discovery_handler_map_clone = discovery_handler_map.clone();
+    let (new_discovery_handler_sender, _): (
+        broadcast::Sender<String>,
+        broadcast::Receiver<String>,
+    ) = broadcast::channel(4);
```

Comment:
nit - How about moving this to a constant that is shared/referenced across all codebase?

Reply:
Like make it a global? The only way to use globals in rust is via the [`lazy_static`](https://crates.io/crates/lazy_static) crate, which I have read has some performance issues and should be avoided if possible. I used it for Prometheus because people may want to add metrics anywhere -- its not predictable.

Reply:
The channel size (of 4)? Sorry, its hard to tell whats highlighted with GitHub.

Reply:
I'm still getting used to Github view too.
yep I meant the (4) channel size. It is used in the discovery handlers as well and if we want to change it later, then it is in one place, plus it makes the code more readable.

Reply:
makes sense! fixed in fa4eba408f57f22a